### PR TITLE
HMOS OHOSChannel Library

### PR DESCRIPTION
**Pull request by priyankabhadrappanavar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,60 +1,66 @@
...
+OHOSChannel is helper library for inter thread communication between main thread and worker thread. 
 
-[![Android Arsenal](https://img.shields.io/badge/Android%20Arsenal-AndroidChannel-green.svg?style=flat)](https://android-arsenal.com/details/1/2547)
+## Source
+Inspired by [skyfe79/AndroidChannel](https://github.com/skyfe79/AndroidChannel)
 
-AndroidChannel is helper library for inter thread communication between main thread and worker thread. AndroidChannel uses HandlerThread for inter thread communication.
+## Feature
+This library is helper library for inter thread communication between main thread and worker thread. 
+OHOSChannel uses HandlerThread for inter thread communication.
```
Comment (samuel-abhishek) [inline README.md]: Replace HandlerThread with HarmonyOS components that were used - EventHandler and EventRunner

File: entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import kr.pe.burt.ohos.lib.ohoschannel.OhosChannel;
+import kr.pe.burt.ohos.lib.ohoschannel.Timer;
+
+/**
+ * MainAbility Class.
+ */
+public class MainAbility extends FractionAbility {
+    public static final HiLogLabel HI_LOG_LABEL = new HiLogLabel(0, 0, "FractionAbility");
+    static final int PING = 0;
+    static final int PONG = 1;
```
Comment (samuel-abhishek) [inline entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java]: Clearly define the scope of the fields within a class

File: ohoschannel/src/main/java/kr/pe/burt/ohos/lib/ohoschannel/OhosChannel.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        };
+        isChannelOpened = true;
+        return true;
+    }
+
+    /**
+     * Close channel.
+     */
+    public void close() {
+        if (isChannelOpened) {
```
Comment (samuel-abhishek) [inline ohoschannel/src/main/java/kr/pe/burt/ohos/lib/ohoschannel/OhosChannel.java]: Incorrect condition!  The function should return if isChannelOpened is false

File: README.md [hunk 4]
```diff
@@ -93,10 +98,9 @@ public boolean onTouchEvent(MotionEvent event) {
 * channel.toUI()
```
Comment (samuel-abhishek) [inline README.md]: The method mentioned here is incorrect. Please check and correct in all the places.

File: entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    protected void onStop() {
+        super.onStop();
+    }
+
+    @Override
+    public boolean onTouchEvent(TouchEvent event) {
+        HiLog.debug(HI_LOG_LABEL, "onTouchEvent " + event.getAction());
+        if (event.getAction() == TouchEvent.PRIMARY_POINT_UP) {
+            if (timer.isAlive()) {
+                textDisplay.setText("1.Touch event detected, counter stopped.");
```
Comment (samuel-abhishek) [inline entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java]: Please remove the index numbers for each statement

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]
Check and handle the review comments

**Review by samuel-abhishek — APPROVED** [review]
Okay

File: entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java (hunk 5 above)
Comment (priyankabhadrappanavar) [reply entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java]: Done

File: README.md (hunk 4 above)
Comment (priyankabhadrappanavar) [reply README.md]: Done

File: ohoschannel/src/main/java/kr/pe/burt/ohos/lib/ohoschannel/OhosChannel.java (hunk 3 above)
Comment (priyankabhadrappanavar) [reply ohoschannel/src/main/java/kr/pe/burt/ohos/lib/ohoschannel/OhosChannel.java]: Done

File: entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java (hunk 2 above)
Comment (priyankabhadrappanavar) [reply entry/src/main/java/kr/pe/burt/ohos/lib/ohos/channel/app/MainAbility.java]: Done

File: README.md (hunk 1 above)
Comment (priyankabhadrappanavar) [reply README.md]: Done

**Review by sharathks84 — APPROVED** [review]
Okay, approved

**Review by Gleelakrishna — APPROVED** [review]
ok

**Pull request merged**